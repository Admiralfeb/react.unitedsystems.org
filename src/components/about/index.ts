export * from './aboutAllies';
export * from './aboutHC';
export * from './aboutRules';
export * from './carriers';
export * from './about';
