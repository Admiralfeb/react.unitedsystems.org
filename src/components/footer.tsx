import Typography from '@material-ui/core/Typography';
import { Button, makeStyles } from '@material-ui/core';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';
import { ReactNode, useContext } from 'react';
import { RealmAppContext } from 'providers';
import { LoginProviders } from 'models/loginProviders';

function ImgCopyright() {
  return (
    <Typography
      variant="body2"
      color="textSecondary"
      style={{ textAlign: 'center' }}
    >
      Thank you to CMDR RaZ uryel for the background image
    </Typography>
  );
}
function Copyright() {
  return (
    <Typography
      variant="body2"
      color="textSecondary"
      style={{ textAlign: 'center' }}
    >
      {'Copyright Â© '}
      <Link color="inherit" href="https://unitedsystems.org/">
        United Systems Cooperative
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100vh',
  },
  main: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(2),
  },
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[200]
        : theme.palette.grey[800],
  },
  center: {
    textAlign: 'center',
  },
}));

export const Footer = (props: { children?: ReactNode | JSX.Element }) => {
  const classes = useStyles();
  const realm = useContext(RealmAppContext);
  const { children } = props;

  return (
    <div className={classes.root}>
      <div className={classes.main}>{children}</div>
      <footer className={classes.footer}>
        <Container maxWidth="sm" className={classes.center}>
          {realm?.checkUserProvider(LoginProviders.userpass) === true && (
            <Button onClick={() => realm.logOut()}>Log out</Button>
          )}
          <Copyright />
          <ImgCopyright />
        </Container>
      </footer>
    </div>
  );
};
