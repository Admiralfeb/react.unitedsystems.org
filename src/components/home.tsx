import { Fade, makeStyles, Paper, Typography } from '@material-ui/core';
import { useDownloadLink } from 'hooks/useDownloadLink';
import {
  lore,
  latin,
  latinTranslation,
  videoID,
  videoPosterID,
} from 'data/home';

const useStyles = makeStyles((theme) => ({
  root: {},
  header: {
    textAlign: 'center',
  },
  paper: {
    textAlign: 'center',
    width: '80%',
    margin: 'auto',
    padding: 5,
    paddingBottom: 10,
    marginBottom: 5,
  },
  latin: {
    display: 'flex',
    flexDirection: 'column',
    textAlign: 'center',
    border: '1px solid white',
    borderRadius: 10,
    width: 350,
    margin: 'auto',
    marginBottom: 10,
  },
  cursive: {
    fontFamily: 'Cinzel, serif',
    fontSize: 32,
  },
  subtitle: {
    flex: '0 0',
  },
  video: {
    width: '100%',
    padding: 0,
  },
  lore: {
    marginBottom: theme.spacing(1),
  },
}));

export const Home = () => {
  const classes = useStyles();
  const { getDownloadLink } = useDownloadLink();
  return (
    <Fade in={true}>
      <div className={classes.root}>
        <Typography variant="h3" className={classes.header}>
          United Systems Cooperative
        </Typography>
        <Paper className={classes.paper}>
          <Typography className={classes.lore}>{lore}</Typography>
          <div className={classes.latin}>
            <Typography className={classes.cursive} variant="h4">
              {latin}
            </Typography>
            <Typography variant="subtitle2" className={classes.subtitle}>
              {latinTranslation}
            </Typography>
          </div>
          <video
            controls
            className={classes.video}
            poster={getDownloadLink(videoPosterID)}
            controlsList="nodownload"
          >
            <source src={getDownloadLink(videoID)} type="video/mp4" />
            "Your browser does not support this video"
          </video>
        </Paper>
      </div>
    </Fade>
  );
};
