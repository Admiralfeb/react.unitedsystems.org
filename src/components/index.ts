export * from './home';
export * from '../routes/information.routes';
export * from './join';
export * from './joinForm';
export * from './notFound';
