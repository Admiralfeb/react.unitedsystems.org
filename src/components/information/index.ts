export * from './aboutAllies';
export * from './aboutHC';
export * from './aboutRules';
export * from './carriers';
export * from './infoSection';
export * from './infographic';
export * from './information';
export * from './miningMap';
export * from './shipReviews';
