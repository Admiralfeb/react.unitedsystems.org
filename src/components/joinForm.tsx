import {
  Button,
  Checkbox,
  Divider,
  FormControl,
  FormControlLabel,
  FormGroup,
  Link,
  makeStyles,
  Paper,
  Radio,
  RadioGroup,
  TextField,
  Typography,
} from '@material-ui/core';
import React, { ChangeEvent, FormEvent, useMemo, useState } from 'react';
import { NavLink } from 'react-router-dom';
import { Controller, useForm } from 'react-hook-form';

const useStyles = makeStyles((theme) => ({
  root: {
    width: 500,
    minWidth: 400,
    margin: 'auto',
  },
  header: {
    textAlign: 'center',
  },
  paper: { padding: theme.spacing(2) },
  question: {
    borderColor: theme.palette.secondary.main,
    borderWidth: 2,
    borderStyle: 'solid',
    borderRadius: 5,
    margin: theme.spacing(1, 0),
    padding: theme.spacing(1),
  },
}));

export const JoinForm = () => {
  const classes = useStyles();
  const { register, handleSubmit, control } = useForm<IJoinFormFields>();
  const [formFields, setFormFields] = useState<IJoinFormFields>({
    cmdr: '',
    discord: '',
    platforms: {
      pc: false,
      xbox: false,
      ps: false,
    },
    playingLength: null,
    reference: null,
    reference2: undefined,
    rules: false,
    timezone: '',
  });
  const [platforms, setPlatforms] = useState<{
    pc: Boolean;
    xbox: Boolean;
    ps: boolean;
  }>({ pc: false, xbox: false, ps: false });

  const ref2Question: string | undefined = useMemo(() => {
    const ref = formFields.reference;
    switch (ref) {
      case 'player':
        return 'Which player referred you?';
      case 'other':
        return "Please explain 'Other'";
      default:
        return undefined;
    }
  }, [formFields]);

  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    const targetName = event.target.name;
    const checked = event.target.checked;
    console.log(targetName);
    if (targetName === 'pc' || targetName === 'xbox' || targetName === 'ps') {
      setFormFields((state) => {
        const platforms = state.platforms;
        platforms[targetName] = checked;
        return { ...state, platforms };
      });
    }
  };

  const onSubmit = (data: IJoinFormFields) => console.log(data);

  return (
    <div className={classes.root}>
      <Typography variant="h3" className={classes.header}>
        Join USC
      </Typography>
      <Paper className={classes.paper}>
        <Typography className={classes.header}>
          Items marked with * are required.
        </Typography>
        <form onSubmit={handleSubmit(onSubmit)}>
          <div className={classes.question}>
            <Typography>
              Please enter your in-game CMDR name without 'CMDR'
            </Typography>
            <TextField
              label="CMDR Name"
              inputRef={register({ required: true, minLength: 2 })}
              name="cmdr"
            />
          </div>
          <div className={classes.question}>
            <Typography>
              Please enter your discord name in format: name#1234
            </Typography>
            <TextField
              label="Discord Name"
              inputRef={register({ required: true, pattern: /^.+#\d{4}$/gi })}
              name="discord"
            />
          </div>
          <div className={classes.question}>
            <Typography>
              Which platform(s) do you play on? Choose all that apply.
            </Typography>
            <FormControl component="fieldset">
              <FormGroup row>
                <FormControlLabel
                  label="PC"
                  control={
                    <Checkbox
                      name="pc"
                      checked={props.value}
                      onChange={(e) => props.onChange(e.target.checked)}
                    />
                  }
                />
                <FormControlLabel
                  label="Xbox One"
                  control={
                    <Checkbox
                      name="xbox"
                      checked={props.value}
                      onChange={(e) => props.onChange(e.target.checked)}
                    />
                  }
                />
                <FormControlLabel
                  label="PS4 / PS5"
                  control={
                    <Checkbox
                      name="ps"
                      checked={props.value}
                      onChange={(e) => props.onChange(e.target.checked)}
                    />
                  }
                />
              </FormGroup>
            </FormControl>
          </div>
          <div className={classes.question}>
            <Typography>How long have you been playing?</Typography>
            <Controller
              as={
                <RadioGroup
                  name="playingLength"
                  value={formFields.playingLength}
                  row
                >
                  <FormControlLabel
                    value="0"
                    control={<Radio />}
                    label="Less than 1 month"
                  />
                  <FormControlLabel
                    value="1"
                    control={<Radio />}
                    label="More than 1 month"
                  />
                  <FormControlLabel
                    value="2"
                    control={<Radio />}
                    label="More than 6 months"
                  />
                  <FormControlLabel
                    value="3"
                    control={<Radio />}
                    label="More than 1 year"
                  />
                </RadioGroup>
              }
              name="playingLength"
              control={control}
              defaultValue="0"
            />
          </div>
          <div className={classes.question}>
            <Typography>How did you find us?</Typography>
            <FormControl component="fieldset" required>
              <RadioGroup name="reference" value={formFields.playingLength} row>
                <FormControlLabel
                  value="reddit"
                  control={<Radio />}
                  label="Reddit"
                />
                <FormControlLabel
                  value="inara"
                  control={<Radio />}
                  label="Inara"
                />
                <FormControlLabel
                  value="player"
                  control={<Radio />}
                  label="Player Referral"
                />
                <FormControlLabel
                  value="facebook"
                  control={<Radio />}
                  label="Facebook"
                />
                <FormControlLabel
                  value="website"
                  control={<Radio />}
                  label="Our Website"
                />
                <FormControlLabel
                  value="forums"
                  control={<Radio />}
                  label="Forums"
                />
                <FormControlLabel
                  value="other"
                  control={<Radio />}
                  label="Other"
                />
              </RadioGroup>
            </FormControl>

            {ref2Question && (
              <div>
                <Divider />
                <FormControl>
                  <Typography>{ref2Question}</Typography>
                  <TextField
                    value={formFields.reference2}
                    onChange={handleInputChange}
                    name="reference2"
                    required
                  />
                </FormControl>
              </div>
            )}
          </div>
          <div className={classes.question}>
            <Typography>What timezone are you in?</Typography>
            <TextField
              value={formFields.timezone}
              onChange={handleInputChange}
              name="timezone"
            />
          </div>
          <div className={classes.question}>
            <Typography>
              I have read and agree to the{' '}
              <Link
                component={NavLink}
                to="/information/about/rules"
                target="_blank"
              >
                rules
              </Link>
              .
            </Typography>
            <FormControlLabel
              control={
                <Checkbox
                  name="rules"
                  inputRef={register({ required: true })}
                />
              }
              label="Yes"
            />
          </div>
          <Button type="submit">Submit Form</Button>
        </form>
      </Paper>
    </div>
  );
};

interface IJoinFormFields {
  cmdr: string;
  discord: string;
  platforms: {
    pc: boolean;
    xbox: boolean;
    ps: boolean;
  };
  playingLength: number | null;
  reference: string | null;
  reference2?: string;
  rules: boolean;
  timezone: string;
}
