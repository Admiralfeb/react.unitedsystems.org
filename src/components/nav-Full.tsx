import { Button, makeStyles } from '@material-ui/core';
import { INavItem } from 'models/navItem';
import { Fragment } from 'react';
import { NavLink } from 'react-router-dom';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    padding: theme.spacing(1),
    backgroundColor: '#424242',
    height: 'auto',
    minHeight: 'auto',
    justifyContent: 'initial',
    zIndex: 1500,
  },
  navLink: {
    backgroundColor: 'transparent',
    height: 'auto',
    color: '#f2f2f2',
    textAlign: 'center',
    textDecoration: 'none',
    // width: theme.spacing(15),
    marginRight: theme.spacing(1),
    fontSize: 18,
  },
  active: {
    backgroundColor: theme.palette.primary.main,
    color: 'white',
  },
  navLinkJoin: {
    backgroundColor: 'transparent',
    border: `${theme.palette.secondary.main} 0.1em solid`,
    color: 'white',
    fontSize: 18,
    '&:hover': {
      backgroundColor: theme.palette.secondary.main,
    },
    width: theme.spacing(15),
  },
  activeJoin: {
    backgroundColor: theme.palette.secondary.main,
  },
  filler: {
    flex: '1 0',
  },
}));

export const NavFull = (props: { navItems: INavItem[] }) => {
  const classes = useStyles();
  const { navItems } = props;

  return (
    <nav className={classes.root}>
      {navItems.map((x) => {
        if (x.to.includes('join')) {
          return (
            <Fragment key={x.to}>
              <div className={classes.filler} />
              <Button
                to={x.to}
                color="secondary"
                className={classes.navLinkJoin}
                activeClassName={classes.activeJoin}
                variant="contained"
                component={NavLink}
              >
                {x.text}
              </Button>
            </Fragment>
          );
        } else {
          return (
            <Button
              key={x.to}
              to={x.to}
              className={classes.navLink}
              activeClassName={classes.active}
              component={NavLink}
            >
              {x.text}
            </Button>
          );
        }
      })}
    </nav>
  );
};
