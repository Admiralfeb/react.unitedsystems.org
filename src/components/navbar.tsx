import { NavLink } from 'react-router-dom';
import { Button, makeStyles } from '@material-ui/core';
import { useLoginAnon } from 'hooks/useLoginAnon';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    padding: '5px 10px',
    backgroundColor: '#333',
    height: 'auto',
    minHeight: 'auto',
    justifyContent: 'initial',
    zIndex: 1500,
  },
  img: {
    height: 'auto',
    width: '55px',
    float: 'left',
    display: 'block',
    marginRight: '1em',
  },
  navLink: {
    backgroundColor: 'transparent',
    color: '#f2f2f2',
    textAlign: 'center',
    textDecoration: 'none',
    marginTop: 'auto',
    marginBottom: 'auto',
    marginLeft: '0.5em',
    textTransform: 'initial',
  },
  active: {
    backgroundColor: theme.palette.primary.main,
    color: 'white',
  },
  filler: {
    flex: '1 0',
  },
}));

export const NavbarComponent = () => {
  const classes = useStyles();
  useLoginAnon();

  return (
    <nav className={classes.root}>
      <Button
        to="/home"
        className={classes.navLink}
        activeClassName={classes.active}
        component={NavLink}
      >
        Home
      </Button>
      <Button
        to="/information"
        exact
        className={classes.navLink}
        activeClassName={classes.active}
        component={NavLink}
      >
        Information
      </Button>
      <Button
        to="/information/builds"
        exact
        className={classes.navLink}
        activeClassName={classes.active}
        component={NavLink}
      >
        USC Builds
      </Button>
      <div className={classes.filler} />
      <Button
        to="/join"
        color="secondary"
        variant="contained"
        component={NavLink}
      >
        Join
      </Button>
    </nav>
  );
};
