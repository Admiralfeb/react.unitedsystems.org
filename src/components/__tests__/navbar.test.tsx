import { cleanup, render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { NavbarComponent } from 'components/navbar';

jest.mock('hooks/useLoginAnon', () => ({
  useLoginAnon: jest.fn(),
}));

let componentBody: RenderResult;

describe('Join', () => {
  beforeEach(() => {
    componentBody = render(
      <MemoryRouter>
        <NavbarComponent />
      </MemoryRouter>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
