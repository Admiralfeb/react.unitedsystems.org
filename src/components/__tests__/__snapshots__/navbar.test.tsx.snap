// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join should match the snapshot 1`] = `
<body>
  <div>
    <nav
      class="makeStyles-root-1"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-3"
        href="/home"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Home
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-3"
        href="/information"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Information
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-navLink-3"
        href="/information/builds"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          USC Builds
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <div
        class="makeStyles-filler-5"
      />
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        href="/join"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Join
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </nav>
  </div>
</body>
`;
